tools/third_party.py

```diff
@@ -208,9 +209,47 @@ def download_from_google_storage(item, bucket):
         env=google_env())
 
 
+# Download the given item from Chrome Infrastructure Package Deployment.
+def download_from_cipd(item):
+    if sys.platform == 'win32':
+        root_dir = "v8/buildtools/win"
+        item += "windows-amd64"
+    elif sys.platform == 'darwin':
+        root_dir = "v8/buildtools/mac"
+        item += "mac-amd64"
+    elif sys.platform.startswith('linux'):
+        root_dir = "v8/buildtools/linux64"
+        item += "linux-amd64"
+
+    # init cipd if necessary
+    if not os.path.exists(path.join(tp(root_dir), ".cipd")):
+        run([
+            tp('depot_tools/cipd'),
+            'init',
+            tp(root_dir),
+            '-force',
+        ],
+            env=google_env())
+
+    # read gn CIPD version from the DEPS file
+    deps = open(tp('v8/DEPS'), 'rb').read()
+    version = re.search("'gn_version': '(git_revision:[0-9a-f]{40})'",
+                        deps).group(1)
```

Comment:
wait, I shouldn't hard code `gn_version` in this is meant to be used for other packages (like clang-format) in the future too